test/integration/webhook/v1alpha2/clusterqueue_test.go

```diff
@@ -128,42 +142,70 @@ var _ = ginkgo.Describe("ClusterQueue Webhook", func() {
 			gomega.Expect(errors.IsForbidden(err)).Should(gomega.BeTrue(), "error: %v", err)
 		})
 
-		ginkgo.It("Should have non-negative quota value when creating", func() {
-			ginkgo.By("Creating a new clusterQueue")
-			cq := testing.MakeClusterQueue("cluster-queue").Resource(
-				testing.MakeResource("cpu").Flavor(testing.MakeFlavor("x86", "-1").Obj()).Obj(),
-			).Obj()
+		ginkgo.DescribeTable("ClusterQueue", func(cq *kueue.ClusterQueue, isForbidden, isInvalid bool) {
 			err := k8sClient.Create(ctx, cq)
-			gomega.Expect(err).Should(gomega.HaveOccurred())
-			gomega.Expect(errors.IsForbidden(err)).Should(gomega.BeTrue(), "error: %v", err)
-		})
-
-		ginkgo.It("Should have quota whose max value is greater than min", func() {
-			ginkgo.By("Creating a new clusterQueue")
-			cq := testing.MakeClusterQueue("cluster-queue").Resource(
+			switch {
+			case isForbidden:
+				gomega.Expect(err).Should(gomega.HaveOccurred())
+				gomega.Expect(errors.IsForbidden(err)).Should(gomega.BeTrue(), "error: %v", err)
+			case isInvalid:
+				gomega.Expect(err).Should(gomega.HaveOccurred())
+				gomega.Expect(errors.IsInvalid(err)).Should(gomega.BeTrue(), "error: %v", err)
+			default:
+				gomega.Expect(err).Should(gomega.Succeed())
+				defer func() {
+					var deleteCQ kueue.ClusterQueue
+					gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(cq), &deleteCQ)).Should(gomega.Succeed())
+					controllerutil.RemoveFinalizer(&deleteCQ, kueue.ResourceInUseFinalizerName)
+					gomega.Expect(k8sClient.Update(ctx, &deleteCQ)).Should(gomega.Succeed())
+					framework.ExpectClusterQueueToBeDeleted(ctx, k8sClient, &deleteCQ, true)
+				}()
+			}
+		},
+			ginkgo.Entry("Should have quota whose max value is greater than min", testing.MakeClusterQueue("cluster-queue").Resource(
```

Comment:
Same for other items

```suggestion
			ginkgo.Entry("Should have quota whose max value is greater than min",
				testing.MakeClusterQueue("cluster-queue").Resource(
```